src/playground-ide.ts

```diff
@@ -147,17 +201,63 @@ export class PlaygroundIde extends LitElement {
       >
       </playground-file-editor>
 
-      <playground-preview
-        part="preview"
-        exportparts="preview-toolbar,
-                     preview-location,
-                     preview-reload-button,
-                     preview-loading-indicator"
-        location="Result"
-        .project=${projectId}
-      ></playground-preview>
+      <div id="rhs">
+        ${this.resizable
+          ? html`<div
+              id="resizeBar"
+              @pointerdown=${this.onResizeBarPointerdown}
+            ></div>`
+          : nothing}
+
+        <playground-preview
+          part="preview"
+          exportparts="preview-toolbar,
+                       preview-location,
+                       preview-reload-button,
+                       preview-loading-indicator"
+          location="Result"
+          .project=${projectId}
+        ></playground-preview>
+      </div>
     `;
   }
+
+  async update(changedProperties: PropertyValues<this>) {
+    if (changedProperties.has('resizable') && this.resizable === false) {
+      // Note we set this property on the RHS element instead of the host so
+      // that when "resizable" is toggled, we don't reset a host value that the
+      // user might have set.
+      this._rhs?.style.removeProperty('--playground-preview-width');
+    }
+    super.update(changedProperties);
+  }
+
+  private onResizeBarPointerdown(event: PointerEvent) {
+    const bar = this._resizeBar;
+    bar.setPointerCapture(event.pointerId);
+
+    const rhsStyle = this._rhs.style;
+    const {left: hostLeft, right: hostRight} = this.getBoundingClientRect();
+    const hostWidth = hostRight - hostLeft;
+    const rhsMinWidth = 100;
+    const rhsMaxWidth = hostWidth - 100;
+
+    const onPointermove = (event: PointerEvent) => {
+      const rhsWidth = Math.min(
+        rhsMaxWidth,
+        Math.max(rhsMinWidth, hostRight - event.clientX)
+      );
+      const percent = (rhsWidth / hostWidth) * 100;
+      rhsStyle.setProperty('--playground-preview-width', `${percent}%`);
+    };
+    bar.addEventListener('pointermove', onPointermove);
+
+    const stopDragging = () => {
+      bar.removeEventListener('pointermove', onPointermove);
+      bar.removeEventListener('pointerup', stopDragging);
```

Comment:
I'm not sure, but should you be [releasing pointer capture](https://developer.mozilla.org/en-US/docs/Web/API/Element/releasePointerCapture)? Not sure if it stays in memory like resize observers after references are GC'd. @justinfagnani any idea?

Reply:
Seems reasonable, done.